### ISSUE TITLE
🚀 Access current `*fiber.Ctx` from template engine

### ISSUE THREAD

**Issue by ayatkyo** [gofiber/fiber#1622]
**Is your feature request related to a problem?**
I want access `*fiber.Ctx` from Template Engine so i can make template function that depend on context like `c.BaseURL()`, `c.Locals()` etc

**Describe the solution you'd like**
Add `fiber.ctx` as `Views.Render` argument

```go
type Views interface {
  Load() error
  Render(*Ctx, io.Writer, string, interface{}, ...string) error
}
```
So we can access context from Template Engine.

For example i add some change to https://github.com/gofiber/template/blob/master/jet/jet.go#L219
```go
// ...
func (e *Engine) Render(c *fiber.Ctx, out io.Writer, template string, binding interface{}, layout ...string) error {
  // ...
  bind := jetVarMap(binding)
  e.Templates.AddGlobal("FiberCtx", c)
  // ...
}
```

so i can access it from template function like snippet below.

**Additional context**
For more detail, i use `Fiber` with `Jet` Template Engine, and this is template function i want to make:
```go
engine.AddFunc("url", jet.Func(func(a jet.Arguments) reflect.Value {
  a.RequireNumOfArguments("url", 1, 1)

  ctx := a.Runtime().Resolve("FiberCtx").Interface().(*fiber.Ctx)
  baseUrl := ctx.BaseURL() + viper.GetString("APP_PATH")

  return reflect.ValueOf(baseUrl + a.Get(0).String())
}))
```

**Comment by da4z** [comment]
I agree, it is necessary

**Comment by ReneWerner87** [comment]
but you could just put the context in the bindings ? currently the render method or template packet is nice and granular and can maybe be used in other contexts without fiber

only dependency that currently exists is the fiber map which is identical to the normal map

since everything is possible with the current variant simple, I see no reason to customize it

**Comment by ReneWerner87** [comment]
next problem is that this change is a breaking change that is not downward compatible

maybe we could provide further methods to add globals which is always present, but i think the bindings are enough

**Comment by ayatkyo** [comment]
Yes that right, this will be huge breaking change and we should avoid that. I have solved this problem using `ctx.locals` based on the answer you gave in https://github.com/gofiber/fiber/issues/1536#issuecomment-922686519

So this issue can be closed.

For those wanting to know the solution, here's what I did.
First we make a Middleware to provide default view data something like this:

```go
app.Use(func(c *fiber.Ctx) error {
  c.Locals("VIEWDATA", fiber.Map{
    "FiberCtx": c,
  })
  return c.Next()
})
```

Then before render function, get view data like this:

```go
app.Get("/", func(c *fiber.Ctx) error {
  data := c.Locals("VIEWDATA").(fiber.Map)
  // add more data to view
  data["another"] = "Data"

  return c.Render("index", data)
})
```

This will avoid breaking change and can be used in current fiber version.

This can be more simplified if PR #1604 get merged, so no need to get locals on every routes.
